core/app/helpers/spree/base_helper.rb

```diff
@@ -175,6 +175,40 @@ def rails_5?
       Rails::VERSION::STRING < '6.0'
     end
 
+    def spree_storefront_product_url(product, options = {})
+      if locale_param
+        options.merge!(locale: locale_param)
+      end
+
+      if frontend_available?
```

Comment:
@MatthewKennedy let's drop off these conditions altogether and use Config value in all cases